### DISCUSSION TITLE
Views - Suggestions

### DISCUSSION THREAD

**Discussion by mjashanks** [Budibase/budibase#652] [Ideas]
Below are a few suggestion improvements for views.

## Filters

1. Attachment column type is currently useless in a view filter, as the operator ( "Equals", "Less Than") has no meaning. Suggest that we add a "Not Empty" - this would be a useful filter for other types as well

2. Date Fields are crrently a bit limited in filters.
- There is no date picker for the value field
- A more common use case would be something like "Last 2 Weeks" or "Last Year"... i.e. dates relavite to the current date, rather than a static date

## Calculate

1. we are currently only allowed to choose "Statistics" - a single valued dropdown. I suggest that we should add a "Count" option, which simply counts all records (and also allows group by). This allows us to produce a count without selecting a number field
2. Should we be able to statistics on a date field ... i.e. Max, Min date ?
3. When we add a Group By, the first column is called "Group". It would be nice if this was the name of the actual column. Not a problem in the backend itself, but it becomes an issue when we use it on the front end - i.e. it feels weird having the select "group" from a column list in charts

**Comment by KernOil** [comment]
we need to be able to remove column in a row that are not associated with that view. currently you can drag the columns out and they will temporally disappear but will all reappear when you leave and re-enter the view. either a bug or a pointless feature currently.
